marquez/client.py

```diff
@@ -0,0 +1,236 @@
+                       nominal_end_time=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}/runs"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "nominalStartTime": nominal_start_time,
+            "nominalEndTime": nominal_end_time,
+            "runArgs": job_run_args
+        }
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job_run(self, job_run_id, **kwargs):
+        path = "jobs/runs/{0}"
+        path_args = [job_run_id]
+        return self.get_request(path, path_args, **kwargs)
+
+    def mark_job_run_running(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.RUN.value, **kwargs)
+
+    def mark_job_run_completed(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.COMPLETE.value, **kwargs)
+
+    def mark_job_run_failed(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.FAIL.value, **kwargs)
+
+    def mark_job_run_aborted(self, job_run_id, **kwargs):
+        return self.mark_job_run_as(
+            job_run_id, JobRunState.ABORT.value, **kwargs)
+
+    @handle_response
+    def mark_job_run_as(self, job_run_id, status, **kwargs):
+        path = "jobs/runs/{0}/{1}".format(job_run_id, status)
+        path_args = [job_run_id, status]
+        payload = None
+        return self.put_request(path, path_args, payload, **kwargs)
+
+    #  Datasets API
+    @handle_response
+    def create_dataset(self, name, datasource_urn, description=None, **kwargs):
+        path = "namespaces/{0}/datasets".format(self.namespace)
+        path_args = [self.namespace]
+        payload = {
+            'name': name,
+            'datasourceUrn': datasource_urn
+        }
+
+        if description:
+            payload['description'] = description
+
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_dataset(self, urn, **kwargs):
+        path = "namespaces/{0}/datasets/{1}"
+        path_args = [self.namespace, urn]
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def list_datasets(self, **kwargs):
+        path = "namespaces/{0}/datasets"
+        path_args = [self.namespace]
+        return self.get_request(path, path_args, **kwargs)
+
+    def put_request(self, path, path_args=None, args=None, **kwargs):
+        path = _compose_path(path, path_args)
+        full_path = "http://{0}/{1}".format(self.marquez_host, path)
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return requests.put(url=full_path, json=args, **kwargs)
+
+    def post_request(self, path, path_args=None, payload=None, **kwargs):
+        path = _compose_path(path, path_args)
+        full_path = "http://{0}/{1}".format(self.marquez_host, path)
+        if 'timeout' not in kwargs:
```

Comment:
Let's make `timeout` an explicit parameter for all http methods.

Reply:
I don't think that's necessary. It's already set by default in the client, and can be overridden for all client calls with an environmental variable. Also, if we make a special case for `timeout`, you could also make the case for many of the other `request` inputs.

Reply:
Actually, we set `timeout` as a global field. Why then access it via `kwargs`?

Reply:
Fixed.